source/1_engine/rex_engine/include/rex_engine/event.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "rex_engine/event_type.h"
+#include "rex_engine/types.h"
+
+namespace rex
+{
+  namespace event_system
+  {
+    struct WindowReizeEvent
+    {
+      s32 window_width;
+      s32 window_height;
+    };
+
+    union Event
```

Comment:
why is this a union? it looks like both these fields will be used at the same time which will be UB at that point.

Reply:
I have not tested it on clang but this is how SDL handles it's events and it seems quite alright here.